cfg.go

```diff
@@ -1,65 +1,35 @@
 package cfg
 
 import (
-	"flag"
-	"log"
-	"os"
-	"strings"
+	"fmt"
 
-	"github.com/joho/godotenv"
 	"github.com/spf13/viper"
 )
 
 const (
-	envModeVar      = "GOENV"
-	localConfigName = "local"
+	configType        = "yaml"
+	defaultConfigName = "default"
 )
 
-var (
-	configPath string
-	configType string
-)
-
-func loadEnvConfig(env string) error {
-	viper.SetConfigName(env)
-	viper.AddConfigPath(configPath)
-	return viper.MergeInConfig()
-}
-
-// Load loads the configuration file depending on the Go environment mode
-func Load(serviceName string) {
-	flag.StringVar(&configPath, "config", "./config", "Configuration dir path")
-	flag.StringVar(&configType, "configtype", "yaml", "Configuration format to use in files")
-	flag.Parse()
-
-	err := godotenv.Load()
-	if err != nil {
-		log.Println("Error loading .env file", err)
-	}
-
+// LoadConfig loads a config directory, the default config file and overrides with a given configuration.
+// It returns a generic configuration, which ideally will be parsed into a struct.
+func LoadConfig(configPath, configName string) (map[string]interface{}, error) {
 	viper.SetConfigType(configType)
-
-	viper.SetEnvPrefix(serviceName)
-	viper.AutomaticEnv()
-	viper.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
-
-	viper.SetConfigName("default")
 	viper.AddConfigPath(configPath)
-	if err := viper.ReadInConfig(); err != nil {
-		log.Println("error reading default configuration file:", err)
-	}
 
-	envConfig, ok := os.LookupEnv(envModeVar)
-	if !ok || envConfig == "" {
-		envConfig = localConfigName
+	// Load default config
+	viper.SetConfigName(defaultConfigName)
+	if err := viper.ReadInConfig(); err != nil {
+		return nil, fmt.Errorf("error loading \"%s\" configuration", defaultConfigName)
```

Comment:
Since the way we will use this library makes it the source of truth we must panic here instead of let the user continue. In other case, I do not see why the pkg will be used for

Reply:
Good idea, I'll work on this.